### PR TITLE
AzureML Designer integration with SAR

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Description

One end 2 end example of Designer integration with SAR

### Checklist:

- [ ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [ ] This PR is being made to `staging` and not `master`.

### CODE REVIEW

File: notebooks/00_quick_start/sar_movieratings_with_azureml_designer.ipynb [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    "One of the features of AzureML Designer is that it is possible for developers to integrate any python library to make it available as a module. In this notebook are are going to show how to integrate [SAR](sar_movielens.ipynb) and several other modules in Designer\n",
+    "\n",
+    "\n",
+    "## Installation\n",
+    "\n",
+    "The first step is to install AzureML Designer SDK. Assuming that you have installed the Recommenders environment `reco_base` as explained in the [SETUP.md](../../SETUP.md), you need to install:\n",
+    "```bash\n",
+    "conda activate reco_base\n",
+    "pip install keyring artifacts-keyring\n",
+    "pip install azureml-designer-tools==0.1.10.post8799295 --extra-index-url=https://msdata.pkgs.visualstudio.com/_packaging/azureml-modules%40Local/pypi/simple/ \n",
```
Comment (dwt0317) [inline notebooks/00_quick_start/sar_movieratings_with_azureml_designer.ipynb]: We published new version of register tool. It supports local folder as module source now. You can try it by: pip install azureml-designer-tools==0.1.18 --extra-index-url=https://msdata.pkgs.visualstudio.com/_packaging/azureml-modules%40Local/pypi/simple/

File: reco_utils/azureml/azureml_designer_modules/entries/score_sar_entry.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import argparse
```
Comment (dwt0317) [inline reco_utils/azureml/azureml_designer_modules/entries/score_sar_entry.py]: Please note score module in this implementation can be deployed as real-time inference. Currently real-time inference has many constraints on custom module. You can take this implementation as example if you're interested https://github.com/hirofumi-s-friends/recommenders/blob/weda/sar_ds_old/azureml-designer-modules/sar/score_sar_entry.py

File: notebooks/00_quick_start/sar_movieratings_with_azureml_designer.ipynb (hunk 1 above)
Comment (miguelgfierro) [reply notebooks/00_quick_start/sar_movieratings_with_azureml_designer.ipynb]: I'll update to the newest version

File: reco_utils/azureml/azureml_designer_modules/entries/score_sar_entry.py (hunk 2 above)
Comment (miguelgfierro) [reply reco_utils/azureml/azureml_designer_modules/entries/score_sar_entry.py]: we decided not to implement this on this PR

**Comment by anargyri** [comment]
Nitpicking: when you say `pip install azureml-designer-tools==0.1.18` is 0.1.18 the latest version? We probably don't want to update this file whenever there is a new release of azureml-designer-tools.

**Comment by miguelgfierro** [comment]
this notebook is focused on that version, which is the one it was tested (in fact also 0.1.10), since designer is still under strong development, we pinned the version. When it is more stable, we can revisit this rule

**Pull request merged**